dotnet/src/Agents/AzureAI/Extensions/AgentDefinitionExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.AI.Projects;
+
+namespace Microsoft.SemanticKernel.Agents.AzureAI;
+
+/// <summary>
+/// Provides extension methods for <see cref="AgentDefinition"/>.
+/// </summary>
+internal static class AgentDefinitionExtensions
+{
+    /// <summary>
+    /// Return the Azure AI tool definitions which corresponds with the provided <see cref="AgentDefinition"/>.
+    /// </summary>
+    /// <param name="agentDefinition">Agent definition</param>
+    /// <exception cref="InvalidOperationException"></exception>
+    public static IEnumerable<ToolDefinition> GetAzureToolDefinitions(this AgentDefinition agentDefinition)
+    {
+        return agentDefinition.Tools.Select<AgentToolDefinition, ToolDefinition>(tool =>
+        {
+            return tool.Type switch
+            {
+                "code_interpreter" => new CodeInterpreterToolDefinition(),
+                "file_search" => new FileSearchToolDefinition(),
+                _ => throw new InvalidOperationException($"Unable to created Azure AI tool definition because of known tool type: {tool.Type}"),
```

Comment:
Suggest NotSupported here as well
Update the exception docs.

```suggestion
                _ => throw new NotSupportedException($"Unable to create Azure AI tool definition because for a not supported tool type: {tool.Type}"),
```